src/controller.rs

```diff
@@ -212,9 +234,19 @@ pub fn grab_pan(
 
         if mouse_button.just_released(drag_button) {
             *ray_hit = None;
+
+            if let Ok(mut primary_window) = primary_window_q.get_single_mut() {
+                primary_window.cursor.visible = true;
+            }
         }
 
         if mouse_button.pressed(drag_button) {
+            if controller.lock_on_drag {
+                if let Ok(mut primary_window) = primary_window_q.get_single_mut() {
+                    primary_window.set_cursor_position(*mouse_pos);
```

Comment:
Use the built-in cursor locking mechanism instead of manually setting the cursor position:

```rust
primary_window.cursor.grab_mode = CursorGrabMode::Locked;
```

Reply:
`CursorGrabMode::Locked` doesn't work like that actually, instead it just confines the mouse cursor to the window, that's why I had to workaround it (https://github.com/bevyengine/bevy/issues/2645). On top of that, it doesn't work on Windows either.

Reply:
Perhaps you are on windows? Windows doesn't support locked mode: https://docs.rs/bevy/latest/bevy/window/struct.Cursor.html#structfield.grab_mode

That said, for consistency, it makes sense to go with your solution.

You have a clippy error - you'll need to ignore the 'too many args' error for `grab_pan`

Reply:
I'm on Arch Linux, but even though that doesn't work for me. Alright, I'll do that!

Reply:
Also, update advanced example to enable both options (follow pattern in the file)